ompi/mca/io/romio341/src/io_romio341_component.c

```diff
@@ -132,18 +132,21 @@ static int register_component(void)
                                            MCA_BASE_VAR_TYPE_INT, NULL, 0, 0,
                                            OPAL_INFO_LVL_9,
                                            MCA_BASE_VAR_SCOPE_READONLY, &delete_priority_param);
+    ompi_io_romio341_version = ROMIO_VERSION_STRING;
```

Comment:
I don't have a problem moving these strings down here (vs. initializing them outside the function), but these MCA params are `MCA_BASE_VAR_SCOPE_READONLY`.  Does that provide any protection here?  Or do `_READONLY` params have the same call-init-multiple-times problem that exists elsewhere?

Reply:
They do have that problem. What is happening is that a new string is allocated and copied and that pointer is returned into the last parameter. This changes from pointing at a static string to an allocated string - which can later be freed and set to NULL. If it's set to NULL, it will be NULL if it happens to hit the register path again and the "default value" will not be set.

Reply:
I wondered about that too, but at the end I assumed the move was done for consistency and style, such that we can easily grep and check in the future.

Reply:
What is done here (and in every other place) is that it gets assigned its default value right before param registration, so no funny business goes on which could alter what the variable is pointing to.